arangodb-net-standard/GraphApi/GraphApiClient.cs

```diff
@@ -1013,5 +1016,127 @@ public virtual async Task<GetEdgeCollectionsResponse> GetEdgeCollectionsAsync(st
+        /// <param name="token">A CancellationToken to observe while waiting for the task to complete or to cancel the task.</param>
+        /// <returns></returns>
+        public virtual async Task<T[]> TraverseGraphAsync<T>(
+            TraverseGraphBody body,
+            CancellationToken token = default)
+        {
+            if (body == null)
+            {
+                throw new ArgumentNullException(nameof(body));
+            }
+
+            if (body.CurrentVertex == null)
+            {
+                throw new Exception($"{nameof(body.CurrentVertex)} cannot be null.");
+            }
+
+            //initialize serialization options
+            var so = new ApiClientSerializationOptions(true, true);
+
+            //build the query string
+            var qs = string.Empty;
+
+            if (body.VertexCollections != null && body.VertexCollections.Count() > 0)
+            {
+                qs += $"WITH {string.Join(",", body.VertexCollections)} {Environment.NewLine}";
+            }
+
+            qs += $"FOR {GetContentString(body.CurrentVertex, so)} ";
+
+            if (body.CurrentEdge != null)
+            {
+                qs += $", {GetContentString(body.CurrentEdge, so)} ";
+            }
+
+            if (body.CurrentPath != null)
+            {
+                qs += $", {GetContentString(body.CurrentPath, so)} ";
+            }
+
+            qs += Environment.NewLine;
+
+            if (body.MinDepth != null)
+            {
+                qs += $"IN {body.MinDepth}";
+                if (body.MaxDepth != null)
+                {
+                    qs += $"..{body.MaxDepth}";
+                }
+                qs += Environment.NewLine;
+            }
+
+            qs += $"{(string.IsNullOrWhiteSpace(body.Direction) ? TraversalDirection.Any : body.Direction)} {body.StartVertex ?? string.Empty} {Environment.NewLine}";
+
+            if (string.IsNullOrWhiteSpace(body.GraphName))
+            {
+                //we're working with a set of edge collections, make sure we have them
+                if (body.EdgeCollections == null || body.EdgeCollections.Count() < 1)
+                {
+                    throw new Exception($"{nameof(body.EdgeCollections)} is required if {nameof(body.GraphName)} is not specified");
+                }
+                else
+                {
+                    qs += $"{string.Join(",", body.EdgeCollections)}{Environment.NewLine}";
+                }
+            }
+            else
+            {
+                //we're working with a named graph
+                qs += $"'{body.GraphName}'{Environment.NewLine}";
+            }
+
+            if (!string.IsNullOrWhiteSpace(body.PruneCondition))
+            {
+                qs += $"PRUNE {body.PruneCondition}{Environment.NewLine}";
+            }
+
+            if (body.Options != null)
+            {
+                qs += $"OPTIONS {GetContentString(body.Options, so)}";
+            }
```

Comment:
Can we move the construction of the query string into a separate function and add some unit tests for it?

Reply:
@mpoeter , I have changed the approach to use fluent interface design pattern to generate the query. I will write tests once we finalize this approach. Thanks :-)